pkg/image/content_helpers.go

```diff
@@ -31,20 +31,64 @@ func fetchFileContentsByPath(ft *filetree.FileTree, fileCatalog *FileCatalog, pa
 func fetchFilesByMIMEType(ft *filetree.FileTree, fileCatalog *FileCatalog, mType string) ([]file.Reference, error) {
 	fileEntries, err := fileCatalog.GetByMIMEType(mType)
 	if err != nil {
-		return nil, fmt.Errorf("unable to fetch file references by MIME type: %w", err)
+		return nil, fmt.Errorf("unable to fetch file references by MIME type (%q): %w", mType, err)
 	}
 
+	// since this query is related to the contents of the path, this should be a strict file ID match
+	return filterCatalogFilesRelativesToTree(ft, fileEntries, true, filetree.FollowBasenameLinks)
+}
+
+// fetchFilesByExtension is a common helper function for resolving file references for a file extension from the file
+// catalog relative to the given tree.
+func fetchFilesByExtension(ft *filetree.FileTree, fileCatalog *FileCatalog, extension string) ([]file.Reference, error) {
+	fileEntries, err := fileCatalog.GetByExtension(extension)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch file references by extension (%q): %w", extension, err)
+	}
+
+	return filterCatalogFilesRelativesToTree(ft, fileEntries, false, filetree.FollowBasenameLinks)
+}
+
+// fetchFilesByBasename is a common helper function for resolving file references for a file basename
+// catalog relative to the given tree.
+func fetchFilesByBasename(ft *filetree.FileTree, fileCatalog *FileCatalog, basename string) ([]file.Reference, error) {
+	fileEntries, err := fileCatalog.GetByBasename(basename)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch file references by basename (%q): %w", basename, err)
+	}
+
+	return filterCatalogFilesRelativesToTree(ft, fileEntries, false, filetree.FollowBasenameLinks)
+}
+
+// fetchFilesByBasenameGlob is a common helper function for resolving file references for a file basename glob pattern
+// catalog relative to the given tree.
+func fetchFilesByBasenameGlob(ft *filetree.FileTree, fileCatalog *FileCatalog, basenameGlob string) ([]file.Reference, error) {
```

Comment:
I know we're trying to move indexes away from `Glob` matching in this PR - Is this the best name here?

The function is written so that `**` and `/` are not supported so I'm trying to think through what kinds of globs are matched then which improve the performance. Let me also check out the test for this since that might help in my review. I'l follow up with other comments as I go through this.